src/tools/ts-generator/linksTSgenerator.cpp

```diff
@@ -0,0 +1,346 @@
+
+#include "linksTSgenerator.h"
+#include <yuni/datetime/timestamp.h>
+
+namespace Antares::TSGenerator
+{
+// ==================
+// Free functions
+// ==================
+std::vector<std::string> extractTargetAreas(fs::path sourceLinkDir)
+{
+    std::vector<std::string> to_return;
+    fs::path pathToIni = sourceLinkDir / "properties.ini";
+    IniFile ini;
+    ini.open(pathToIni); // gp : we should handle reading issues
+    for (auto* s = ini.firstSection; s; s = s->next)
+    {
+        std::string targetAreaName = transformNameIntoID(s->name);
+        to_return.push_back(targetAreaName);
+    }
+    return to_return;
+}
+
+bool pairs_match(const LinkPair& p1, const LinkPair& p2)
+{
+    return (p1.first == p2.first && p1.second == p2.second)
+           || (p1.first == p2.second && p1.second == p2.first);
+}
+
+const LinkPair* getMatchingPairInCollection(const LinkPair& pair, const LinkPairs& collection)
+{
+    for(const auto& p : collection)
+    {
+        if (pairs_match(pair, p))
+            return &p;
+    }
+    return nullptr;
+}
+
+bool readLinkGeneralProperty(StudyParamsForLinkTS& params,
+                             const Yuni::String& key,
+                             const Yuni::String& value)
+{
+    if (key == "derated")
+    {
+        return value.to<bool>(params.derated);
+    }
+    if (key == "nbtimeserieslinks")
+    {
+        return value.to<unsigned int>(params.nbLinkTStoGenerate);
+    }
+    if (key == "seed-tsgen-links")
+    {
+        unsigned int seed {0};
+        if (! value.to<unsigned int>(seed))
+            return false;
+        params.random.reset(seed);
+        return true;
+    }
+    return true; // gp : should we return true here ?
+}
+
+std::vector<LinkTSgenerationParams> CreateLinkList(const LinkPairs& linksFromCmdLine)
+{
+    std::vector<LinkTSgenerationParams> to_return(linksFromCmdLine.size());
+    // gp : following loop should be improved : we shouldn't need an index
+    unsigned int index = 0;
+    for (const auto& link_pair : linksFromCmdLine)
+    {
+        to_return[index].namesPair = link_pair;
+        index++;
+    }
+    return to_return;
+}
+
+LinkTSgenerationParams* findLinkInList(const LinkPair& link_to_find,
+                                       std::vector<LinkTSgenerationParams>& linkList)
+{
+    for(auto& link : linkList)
+    {
+        if (link.namesPair == link_to_find)
```

Comment:
why don't you re-use pairs_match

Reply:
Good point ! `pairs_match` is a bit less strict than `pair<>::operator==`, since it returns `true` if `a.first==b.second && a.second == b.first` (not the case with `pair<>::operator==`)